src/game/Entities/Unit.cpp

```diff
@@ -10631,8 +10631,11 @@ void Unit::CleanupsBeforeDelete()
         ClearComboPointHolders();
         RemoveAllAuras(AURA_REMOVE_BY_DELETE);
     }
-    if (IsVehicle())
+    if (m_vehicleInfo)
+    {
         m_vehicleInfo->Cleanup();
+        m_vehicleInfo = nullptr;
```

Comment:
I believe the nullptr is unnecessary. It will happen when unit is deleted

Reply:
I thought so too, and then I had a crash
![image](https://github.com/user-attachments/assets/80323d88-43df-40df-bc42-370c460ac80b)

Reply:
Well, I mean sure, that fails due to MANGOS_ASSERT(m_passengers.empty()); but I fail to see how calling destructor (by setting the ptr to nullptr) fixes that.

Reply:
Because setting it to nullptr early means that `IsVehicle()` or `if (m_vehicleInfo)` returns false, both of which are used in other aspects of the cleanup.

I haven't yet been able to provoke this same crash again, the server has been running stably for ~2 hours now with no crashes since I added the `= nullptr;` and I'll keep it running some more to see if maybe the crash re-occurs. If it doesn't then I'll just assume setting it to `nullptr` fixed the issue. If it does I'll have to investigate some more. In any case this crash occurred as shown in the crashlog and I couldn't find any other reason for why passengers might not be empty when the crash occurred